lib/Client.php

```diff
@@ -183,21 +193,24 @@ public function getFeatureFlag(
                         $personProperties,
                         $groupProperties
                     );
-
                 } catch (InconclusiveMatchException $e) {
                     $result = null;
+                } catch (Exception $e) {
+                    $result = null;
+                    error_log("[PostHog][Client] Error while computing variant:" . $e->getMessage());
                 }
-    
             }
         }
 
+        $flagWasEvaluatedLocally = !is_null($result);
 
-        if (is_null($result)) {
+        if (!$flagWasEvaluatedLocally && !$onlyEvaluateLocally) {
             try {
                 $featureFlags = $this->fetchFeatureVariants($distinctId, $groups, $personProperties, $groupProperties);
                 $result = $featureFlags[$key] ?? $defaultValue;
```

Comment:
I assume you're about to follow up with a PR that removes defaultValue, in which case okay to do this then! Publish the version only after this change makes it in, since its breaking as well 😅 

```suggestion
                $result = $featureFlags[$key];
```